tests/v2/validation/rbac/rbac_test.go

```diff
@@ -0,0 +1,260 @@
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	provisioning "github.com/rancher/rancher/tests/v2/validation/provisioning"
+	log "github.com/sirupsen/logrus"
+)
+
+const roleOwner = "cluster-owner"
+const roleMember = "cluster-member"
+
+type rbTestSuite struct {
+	suite.Suite
+	client             *rancher.Client
+	standardUser       *management.User
+	standardUserClient *rancher.Client
+	session            *session.Session
+	cluster            *management.Cluster
+	adminProject       *management.Project
+}
+
+
+func (rb *rbTestSuite) TearDownSuite() {
+	rb.session.Cleanup()
+}
+
+func (rb *rbTestSuite) SetupSuite() {
+	testSession := session.NewSession(rb.T())
+	rb.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(rb.T(), err)
+
+	rb.client = client
+
+	//Get cluster name from the config file and append cluster details in rb
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(rb.T(), clusterName, "Cluster name to install should be set")
+	clusterID, err := clusters.GetClusterIDByName(rb.client, clusterName)
+	require.NoError(rb.T(), err, "Error getting cluster ID")
+	rb.cluster, err = rb.client.Management.Cluster.ByID(clusterID)
+	require.NoError(rb.T(), err)
+
+}
+
+func(rb *rbTestSuite) ValidateListCluster(role string){
+	
+	//Testcase1 Verify cluster members - Owner/member are able to list clusters
+	clusterList,err := listClusters(rb.standardUserClient,)
+	require.NoError(rb.T(), err)
+	assert.Equal(rb.T(), 1, len(clusterList.Data))
+	assert.Equal(rb.T(), rb.cluster.ID, clusterList.Data[0].Status.ClusterName )
+}
+
+func (rb *rbTestSuite) ValidateListProjects(role string) {
+
+	//Testcase2 Verify members of cluster are able to list the projects in a cluster
+	//Get project list as an admin
+	projectlistAdmin, err := listProjects(rb.client, rb.cluster.ID)
+	require.NoError(rb.T(), err)
+	//Get project list as a cluster owner/member
+	projectlistClusterMembers, err := listProjects(rb.standardUserClient, rb.cluster.ID)
+	require.NoError(rb.T(), err)
+	switch role {
+	case roleOwner:
+		//assert length of projects list obtained as an admin and a cluster owner are equal
+		assert.Equal(rb.T(), len(projectlistAdmin), len(projectlistClusterMembers))
+		//assert projects values obtained as an admin and the cluster owner are the same
+		assert.Equal(rb.T(), projectlistAdmin, projectlistClusterMembers)
+	case roleMember:
+			//assert projects list obtained as a cluster member is empty
+			assert.Equal(rb.T(), 0, len(projectlistClusterMembers))
+	}
+}
+
+	func (rb *rbTestSuite) ValidateCreateProjects(role string) {
+
+	//Testcase3 Validate if cluster members can create a project in the downstream cluster
+	createProjectAsClusterMembers, err := createProject(rb.client, rb.cluster.ID)
+	require.NoError(rb.T(), err)
+	log.Info("Created project as a ",role, " is ",createProjectAsClusterMembers.Name)
```

Comment:
I think logging and comments are really explanatory, there's only one thing I want to point out, but I don't believe that this is necessary. Go test has its own logging too, so in the tests, this could also be used `rb.T().Logf("Created project as a %v is %v", role, createProjectAsClusterMembers.Name)`

Reply:
Sure. thanks Ugur! I haven't changed the current logging for the validation calls but I used it in the TestRbac call.